### ISSUE TITLE
Bug: missing Jenkins secrets when jx create cluster gke with --kaniko option

### ISSUE THREAD

**Issue by igdianov** [jenkins-x/jx#3047]
### Summary 

When creating new GKE cluster with `--kaniko` option, `jx create cluster gke` command does not initialize Jenkins secrets, i.e. `jenkins-x-chartmuseum`, `jx-pipeline-git-github-github` do not exist in `Jenkins --> Credentials` after successful installation. 

### Steps to reproduce the behavior

1. In the GCP Console, go to the Manage resources page and select or create a project: https://console.cloud.google.com/cloud-resource-manager
Note: If you don't plan to keep the resources you create in this tutorial, create a new project instead of selecting an existing project. After you finish, you can delete the project, removing all resources associated with the project and tutorial.

2. Make sure that billing is enabled for your project: https://cloud.google.com/billing/docs/how-to/modify-project

3. Follow instructions from https://cloud.google.com/container-registry/docs/quickstart to enable Container Registry API for your project.

4. Follow instructions from https://cloud.google.com/container-registry/docs/get-image-vulnerabilities if you want to enable Container Analysis API for your project

5. Open Cloud Shell for selected project:

```sh
export CLUSTER_NAME=kaniko-test

jx create cluster gke \
--project-id=$DEVSHELL_PROJECT_ID \
--docker-registry=gcr.io/$DEVSHELL_PROJECT_ID \
--cluster-name=$CLUSTER_NAME \
--machine-type='n1-standard-4' \
--min-num-nodes='2' \
--preemptible \
--enhanced-apis \
--enhanced-scopes \
--kaniko \
--skip-login
```

### Expected behavior

Jenkins secrets are provisioned

### Actual behavior

Jenkins secrets are empty, the pipelines fail to run without `jenkins-x-chartmuseum`, `jx-pipeline-git-github-github` secrets.

![image](https://user-images.githubusercontent.com/20428629/52543606-eb0ab000-2d5f-11e9-8bb8-e2bafc4f6d3d.png)

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 1.3.867
jenkins x platform 0.0.3368
Kubernetes cluster v1.11.6-gke.2
kubectl            v1.10.7
helm client        v2.12.3+geecf22f
helm server        v2.12.3+geecf22f
git                git version 2.11.0
Operating System   Debian GNU/Linux 9.6 (stretch)
```

### Jenkins type

- [x] Classic Jenkins
- [ ] Serverless Jenkins

### Kubernetes cluster

GKE created with `jx create cluster gke` command above.

### Operating system / Environment

Google Cloud Shell `>_`

**Comment by garethjevans** [comment]
Hi @igdianov, can you confirm if the secrets are missing from the cluster or just not viewable from Jenkins?

e.g. `kubectl get secret jenkins-x-chartmuseum`

**Comment by igdianov** [comment]
@garethjevans I installed new GKE cluster with `--kaniko` option. The  `jenkins-x-chartmuseum` secret exists. The Jenkins Credentials are not populated. 

```sh
$ kubectl get secret jenkins-x-chartmuseum -o yaml
apiVersion: v1
data:
  BASIC_AUTH_PASS: XXXXXXXXXXXXXXXXXXXXXXXX=
  BASIC_AUTH_USER: YWRtaW4=
kind: Secret
metadata:
  annotations:
    jenkins.io/credentials-keybinding-password: BASIC_AUTH_PASS
    jenkins.io/credentials-keybinding-username: BASIC_AUTH_USER
  creationTimestamp: 2019-02-12T22:23:33Z
  labels:
    app: chartmuseum
    chart: chartmuseum-1.1.2
    heritage: Tiller
    jenkins.io/credentials-type: usernamePassword
    release: jenkins-x
  name: jenkins-x-chartmuseum
  namespace: jx
  resourceVersion: "1940"
  selfLink: /api/v1/namespaces/jx/secrets/jenkins-x-chartmuseum
  uid: d54021eb-2f14-11e9-8941-42010a8a0044
type: Opaque
```
![image](https://user-images.githubusercontent.com/20428629/52672798-b0c21f80-2ed3-11e9-8b03-8d72a0ccc1d5.png)

**Comment by igdianov** [comment]
@garethjevans Cluster with `--kaniko` is missing `jx-pipeline-git-github-github` secret. Here is the list of secrets in the cluster :

```sh
$ kubectl get secrets
NAME                                           TYPE                                  DATA      AGE
cleanup-token-2m95k                            kubernetes.io/service-account-token   3         5h
default-token-pkmp7                            kubernetes.io/service-account-token   3         5h
expose-token-5gjjd                             kubernetes.io/service-account-token   3         5h
jenkins                                        Opaque                                4         5h
jenkins-docker-cfg                             Opaque                                1         5h
jenkins-maven-settings                         Opaque                                1         5h
jenkins-npm-token                              Opaque                                1         5h
jenkins-release-gpg                            Opaque                                4         5h
jenkins-ssh-config                             Opaque                                1         5h
jenkins-token-jgcl6                            kubernetes.io/service-account-token   3         5h
jenkins-x-chartmuseum                          Opaque                                2         5h
jenkins-x-controllercommitstatus-token-c6rxk   kubernetes.io/service-account-token   3         5h
jenkins-x-controllerrole-token-mt6js           kubernetes.io/service-account-token   3         5h
jenkins-x-controllerteam-token-hzn4z           kubernetes.io/service-account-token   3         5h
jenkins-x-controllerworkflow-token-jz7tl       kubernetes.io/service-account-token   3         5h
jenkins-x-docker-registry-secret               Opaque                                1         5h
jenkins-x-gcactivities-token-qv686             kubernetes.io/service-account-token   3         5h
jenkins-x-gcpods-token-pfxcj                   kubernetes.io/service-account-token   3         5h
jenkins-x-mongodb                              Opaque                                1         5h
jx-basic-auth                                  Opaque                                1         5h
jx-install-config                              Opaque                                2         5h
kaniko-secret                                  Opaque                                1         5h
nexus                                          Opaque                                1         5h
```

Here is list from another cluster created without `--kaniko` option:

```sh
kubectl get secrets
NAME                                           TYPE                                  DATA      AGE
cleanup-token-rssn7                            kubernetes.io/service-account-token   3         2d
default-token-4ct5h                            kubernetes.io/service-account-token   3         2d
expose-token-crmpf                             kubernetes.io/service-account-token   3         2d
jenkins                                        Opaque                                4         2d
jenkins-docker-cfg                             Opaque                                1         2d
jenkins-maven-settings                         Opaque                                1         2d
jenkins-npm-token                              Opaque                                1         2d
jenkins-release-gpg                            Opaque                                4         2d
jenkins-ssh-config                             Opaque                                1         2d
jenkins-token-q5tpr                            kubernetes.io/service-account-token   3         2d
jenkins-x-chartmuseum                          Opaque                                2         2d
jenkins-x-controllercommitstatus-token-qp6kr   kubernetes.io/service-account-token   3         2d
jenkins-x-controllerrole-token-9q88j           kubernetes.io/service-account-token   3         2d
jenkins-x-controllerteam-token-tnq7z           kubernetes.io/service-account-token   3         2d
jenkins-x-controllerworkflow-token-6dg6v       kubernetes.io/service-account-token   3         2d
jenkins-x-docker-registry-secret               Opaque                                1         2d
jenkins-x-gcactivities-token-nfbpt             kubernetes.io/service-account-token   3         2d
jenkins-x-gcpods-token-h4nbt                   kubernetes.io/service-account-token   3         2d
jenkins-x-mongodb                              Opaque                                1         2d
jx-basic-auth                                  Opaque                                1         2d
jx-install-config                              Opaque                                2         2d
jx-pipeline-git-github-github                  Opaque                                2         2d
nexus                                          Opaque                                1         2d
```

**Comment by wenzlaff** [comment]
Seems to be related to the latest jenkins image jenkinsxio/jenkinsx:0.45 
In 0.43 the secrets are available, when switching to 0.45 not.
